python_modules/dagster/dagster_tests/daemon_tests/test_dagster_daemon.py

```diff
@@ -51,3 +54,21 @@ def test_ephemeral_instance():
     runner = CliRunner()
     with pytest.raises(Exception, match="DAGSTER_HOME is not set"):
         runner.invoke(run_command, env={"DAGSTER_HOME": ""}, catch_exceptions=False)
+
+
+def test_daemon_json_logs(
+    monkeypatch: pytest.MonkeyPatch,
+    caplog: pytest.LogCaptureFixture,
+) -> None:
+    # https://github.com/pytest-dev/pytest/issues/2987#issuecomment-1460509126
+    monkeypatch.setattr(caplog.handler, "formatter", get_structlog_json_formatter())
```

Comment:
might be good to leave a comment just explaining why this is relevant to the test, in addition to the issue.